File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        dict = { '(': ')', '{': '}', '[': ']' }
        stack = []

        for c in s:
            if c in dict:
                stack.append(c)
                
            elif len(stack) == 0 or dict[stack.pop()] != c:
                return False

        return len(stack) == 0

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let stack = [];
    charMap = { ')': '(', '}':'{', ']': '[' }
    
    for (let c of s) {
        if (c in charMap) {
            let curr = stack.pop();
            if (curr !== charMap[c])
                return false;
        } else {
            stack.push(c);
        }
    }

    return (stack.length === 0 ? true : false);
};